azure/scope/cluster.go

```diff
@@ -845,8 +853,29 @@ func (s *ClusterScope) AdditionalTags() infrav1.Tags {
 	return tags
 }
 
-// APIServerPort returns the APIServerPort to use when creating the load balancer.
-func (s *ClusterScope) APIServerPort() int32 {
+// KCPBindPort returns the port that kube-apiserver will listen on for each individual control plane node.
+func (s *ClusterScope) KCPBindPort() (int32, error) {
+	if s.Cluster.Spec.ControlPlaneRef != nil && s.Cluster.Spec.ControlPlaneRef.Kind == "KubeadmControlPlane" {
```

Comment:
While revisiting past discussions on the topic, I came across [this old comment](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3063#discussion_r781680104) (which I wrote 🤦‍♀️) which still stands. I think the way this PR does it is a bit better because we don't assume Kind to be `KubeadmControlPlane` and don't fail if it's not the case (just preserve the existing behavior) but we need to keep in mind we can't assume a specific control plane provider.